test/unit/uri_parser_tests.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const parseConnectionString = require('../../lib/core/uri_parser');
+const expect = require('chai').expect;
+
+describe('New URL Parser', function() {
+  /**
+   * @ignore
+   */
+  it('should return the correct hostname when given a URL with srv', function(done) {
+    parseConnectionString(
+      'mongodb+srv://admin:password@compass-data-sets-e06dc.mongodb.net/admin',
```

Comment:
This test is working because `compass-data-sets-e06dc.mongodb.net` host exists. If for some reason this host won't be available in the future, the test will fail.
@mbroadst this is what I fixed here: https://github.com/mongodb-js/connection-model/pull/196
Pls let me know what do you think about it.

Reply:
Actually this test only checks the parsing so it doesn't matter if the host exists or not.

Reply:
`parseConnectionString` function checks if URI is a `srv record`. If it is, it calls `parseSrvConnectionString`, which has `dns.resolveSrv` and `dns.resolveTxt` methods that will fail for unexacting hosts.
Try to replace URI that you have in this test with for example this one:
`mongodb+srv://admin:password@test-data-sets-e06dc.mongodb.net/admin`
It won't work.